superdesk/types/archive.py

```diff
@@ -0,0 +1,453 @@
+    qcode: str
+    name: str
+    scheme: str | None = None
+
+
+@dataclass
+class Genre:
+    qcode: str
+    name: str
+
+
+@dataclass
+class CompanyCodes:
+    qcode: str
+    name: str
+    security_exchange: str
+
+
+@dataclass
+class Slugline:
+    phrase: str
+    keyword: str
+    text: str
+
+
+class ContentStates(Enum):
+    DRAFT = "draft"
+    INGESTED = "ingested"
+    ROUTED = "routed"
+    FETCHED = "fetched"
+    SUBMITTED = "submitted"
+    IN_PROGRESS = "in_progress"
+    SPIKED = "spiked"
+    PUBLISHED = "published"
+    KILLED = "killed"
+    CORRECTED = "corrected"
+    SCHEDULED = "scheduled"
+    RECALLED = "recalled"
+    UNPUBLISHED = "unpublished"
+    CORRECTION = "correction"
+    BEING_CORRECTED = "being_corrected"
+
+
+class PubStatuses(Enum):
+    USABLE = "usable"
+    HOLD = "withheld"
+    CANCELED = "canceled"
+
+
+@dataclass
+class Author:
+    uri: str
+    parent: str
+    name: str
+    role: str
+    jobtitle: str
+    sub_label: str
+
+
+@dataclass
+class Geopoint:
+    lat: float
+    lon: float
+
+
+@dataclass
+class GroupRefRefs:
+    idRef: str
+    _id: str
+    uri: str
+    guid: str
+    type: str
+    location: str
+    headline: str
+    slugline: str
+
+
+@dataclass
+class GroupRef:
+    id: str
+    refs: GroupRefRefs
+
+
+@dataclass
+class Place:
+    code: str
+    name: str
+    qcode: str
+    scheme: str
+    feature_class: str
+    location: list[Geopoint]
+    continent_code: str | None = None
+    region: str | None = None
+    region_code: str | None = None
+    locality: str | None = None
+    state: str | None = None
+    country: str | None = None
+    world_region: str | None = None
+    locality_code: str | None = None
+    state_code: str | None = None
+    country_code: str | None = None
+    world_region_code: str | None = None
+    rel: str | None = None
+    tz: str | None = None
+
+
+@dataclass
+class Ref:
+    _id: str
+    key: str
+    uri: str
+    guid: str
+    type: str
+    source: str | None = None
+
+
+@dataclass
+class Located:
+    state_code: str
+    city: str
+    tz: str
+    country_code: str
+    dateline: str
+    alt_name: str
+    state: str
+    city_code: str
+    country: str
+    code: str
+    scheme: str
+    location: list[Geopoint]
+    place: Place | None = None
+
+
+@dataclass
+class Dateline:
+    source: str
+    located: Located | None = None
+    date: datetime | None = None
+    text: str | None = None
+
+
+@dataclass
+class POI:
+    x: float
+    y: float
+
+
+@dataclass
+class FeatureMedia:
+    _id: str
+    guid: str
+    unique_id: int
+
+
+@dataclass
+class Associations:
+    featuremedia: FeatureMedia | None = None
+
+
+@dataclass
+class EntityMetadataItem:
+    name: str
+    qcode: str
+    scheme: str
+    source: str
+
+
+@dataclass
+class EntityMetadata:
+    metadata: list[EntityMetadataItem] | None = None
+
+
+@dataclass
+class MarkedDesk:
+    desk_id: Annotated[str | None, validate_data_relation_async("desks")] = None
```

Comment:
Same here, should probably use fields.ObjectId. In fact, anything that uses `Resource.rel` in the old schema, where the `type` param is not changed, should use fields.ObjectId